test/v1/controllers/requestParsers/validators/RetrieveBalanceValidatorSpec.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.requestParsers.validators
+
+import support.UnitSpec
+import v1.models.errors.NinoFormatError
+import v1.models.request.retrieveBalance.RetrieveBalanceRawRequest
+import v1.fixtures.RetrieveBalanceFixture._
+
+class RetrieveBalanceValidatorSpec extends UnitSpec {
+
+  val validator = new RetrieveBalanceValidator()
+
+  "running a validation" should {
+    "return no errors" when {
+      "a valid request is supplied" in {
+        validator.validate(RetrieveBalanceRawRequest(validNino)) shouldBe Nil
+      }
+    }
+
+    "return NinoFormatError error" when {
+      "an invalid nino is supplied" in {
+        validator.validate(RetrieveBalanceRawRequest(invalidNino)) shouldBe List(NinoFormatError)
```

Comment:
These are the same models as used in the parser spec, and so could use the same vals.